PG-1113 Improve readme (#324)
percona/pg_tde: README.md

## Table of contents
1. [Overview](#overview)
2. [Documentation](#documentation)
1. [Percona Server for PostgreSQL](#percona-server-for-postgresql)
3. [Build from sources](#building-from-sources-for-community-postgresql)
4. [Run in docker](#run-in-docker)
5. [Setting up](#setting-up)
6. [Helper functions](#helper-functions)

## Overview
Transparent Data Encryption offers encryption at the file level and solves the problem of protecting data at rest. The encryption is transparent for users allowing them to access and manipulate the data and not to worry about the encryption process. As a key provider, the extension supports the keyringfile and  [Hashicorp Vault](https://www.vaultproject.io/).

### This extension provides two `access methods` with different options:

#### `tde_heap_basic` access method
- Works with community PostgreSQL 16 and 17 or with [Percona Server for PosgreSQL 17](https://docs.percona.com/postgresql/17/postgresql-server.html)
- Encrypts tuples and WAL
- **Doesn't** encrypt indexes, temporary files, statistics
- CPU expensive as it decrypts pages each time they are read from bufferpool

#### `tde_heap` access method

> - Works only with [Percona Server for PosgreSQL 17](https://docs.percona.com/postgresql/17/postgresql-server.html)

Reviewer: typo in PostgreSQL

Author: Fixed

# pg_tde: Transparent Database Encryption for PostgreSQL

> `Experimental` PostgreSQL extension that provides data encryption at rest on table level. [We need your feedback!](https://github.com/percona/pg_tde/discussions/151)

Reviewer 2: What if we reword it as follows:
The PostgreSQL extension that provides data encryption at rest on the table level. It is in the experimental stage and is actively developed. We need your feedback!

Correct me if I understood the message wrong

Author: Agree, changed

> The PostgreSQL extension that provides data encryption at rest on the table level. It is in the experimental stage and is actively developed. [We need your feedback!](https://github.com/percona/pg_tde/discussions/151)

Reviewer: I would propose to rewrite: 
"The PostgreSQL extension provides data at rest encryption on table level. It is currently in an experimental phase and is under active development."

Author: Changed

## Table of contents
1. [Overview](#overview)
2. [Documentation](#documentation)
3. [Installation](#installation)
   1. [Percona Server for PostgreSQL](#install-pecona-server-for-postgresql-which-includes-pg_tde-with-package-manager)
   2. [With package manager](#install-pg_tde-extension-with-package-manager)
   3. [From sources](#build-and-install-pg_tde-extension-from-sources)
   4. [Run in docker](#run-in-docker)
4. [Setting up](#setting-up)
5. [Helper functions](#helper-functions)

## Overview

> Transparent Data Encryption offers encryption at the file level and solves the problem of protecting data at rest. The encryption is transparent for users allowing them to access and manipulate the data and not to worry about the encryption process. As a key provider extension supports file and [Hashicorp Vault](https://www.vaultproject.io/).

Reviewer 2: ```suggestion
Transparent Data Encryption offers encryption at the file level and solves the problem of protecting data at rest. The encryption is transparent for users allowing them to access and manipulate the data and not to worry about the encryption process. As a key provider, the extension supports the keyringfile and [Hashicorp Vault](https://www.vaultproject.io/).
```

Author: Done

> Transparent Data Encryption offers encryption at the file level and solves the problem of protecting data at rest. The encryption is transparent for users allowing them to access and manipulate the data and not to worry about the encryption process. As a key provider, the extension supports the keyringfile and  [Hashicorp Vault](https://www.vaultproject.io/).

Reviewer: Above you've mentioned table level and now file level. Can be confusing I think. 

also "to access and to manipulate..."

Author: Updated title. Regarding this `to something` I checked it with Anastasia and it seems that one `to` is enough in this case.

### This extension provides two `access methods` with different options:

#### `tde_heap_basic` access method
- Works with community PostgreSQL 16 and 17
- Encrypts tuples and WAL
- **Doesn't** encrypt indexes, temporary files, statistics

> - CPU expensive as it decrypts pages each time they read from bufferpool 

Reviewer 2: ```suggestion
- CPU expensive as it decrypts pages each time they are read from bufferpool 
```

#### `tde_heap` access method
- Works only with [Percona Server for PosgreSQL 17](https://docs.percona.com/postgresql/17/postgresql-server.html)
- Uses extended Storage Manager and WAL APIs

> - Encrypts tupes, WAL and indexes

Reviewer 2: ```suggestion
- Encrypts tuples, WAL and indexes
```

Author: Changed

- **Doesn't** encrypt temporary files and statistics **yet**
- Faster and cheaper than `tde_heap_basic`

## Documentation

> Full and comprehensive documentation about `pg_tde` available at https://percona.github.io/pg_tde/.

Reviewer 2: ```suggestion
Full and comprehensive documentation about `pg_tde` is available at https://percona.github.io/pg_tde/.
```

Author: Done

## Installation

You can choose one of the given options.

> ### Install Pecona Server for PostgreSQL which includes `pg_tde` with package manager

Reviewer 2: ```suggestion
### Install Percona Server for PostgreSQL which includes `pg_tde` with the package manager
```

Author: Done

   1. Install [percona-release](https://docs.percona.com/percona-software-repositories/installing.html) tool to configure repositories

>    2. Enable Percona Distribuition for PostgreSQL repository  

Reviewer: That is correct for now you need the experimental repo to get the pg_tde extension. After the beta release, it will be part of the official repository and we need to change this part here again.

        ```sh
        sudo percona-release enable-only ppg-17
        ```
   3. Install Percona Server for PosrgreSQL
   - For Debian and Ubuntu
        ```sh
        sudo apt update
        sudo apt install percona-ppg-server-17
        ```
   - For RHEL 8 compatible OS
        ```sh
        sudo yum install percona-ppg-server17

>         ```

Reviewer 2: I would add a step to install percona-postgresql-17-pg-tde here. And rename the next section to show how to install pg_tde for Community PostgreSQL / PPG 16

Author: Removed this section at all

        ```sh
        sudo yum install percona-pg_tde_XX
        ```

### Build and install `pg_tde` extension from sources
  1. Install required dependencies (replace XX with 16 or 17)
   - On Debian and Ubuntu:
        ```sh
        sudo apt install make gcc autoconf libcurl4-openssl-dev postgresql-server-dev-XX
        ```
     
   - On RHEL 8 compatible OS:
        ```sh
        sudo yum install epel-release
        yum --enablerepo=powertools install git make gcc autoconf libcurl-devel perl-IPC-Run redhat-rpm-config openssl-devel postgresqlXX-devel 
        ```

   - On MacOS:
        ```sh
        brew install make autoconf curl gettext postresql@XX
        ```

  2. Install or build postgresql 16 or 17
  3. If postgres is installed in a non standard directory, set the `PG_CONFIG` environment variable to point to the `pg_config` executable

  4. Clone the repository, build and install it with the following commands:  

     ```sh
     git clone https://github.com/percona/pg_tde
     ```
  
   5. Compile and install the extension

      ```sh
      cd pg_tde
      ./configure
      make USE_PGXS=1
      sudo make USE_PGXS=1 install
      ```

### Run in Docker

> There is a [docker image](https://hub.docker.com/r/perconalab/pg_tde) with `pg_tde` based community [PostgreSQL 16](https://hub.docker.com/_/postgres) 

Reviewer: Don't we have the image for PG 17 as well?

Author: Nope, only for 16: https://github.com/percona/pg_tde/blob/main/docker/Dockerfile

```
docker run --name pg-tde -e POSTGRES_PASSWORD=mysecretpassword -d perconalab/pg_tde
```

> Docker file available [here](https://github.com/percona/pg_tde/blob/main/docker/Dockerfile)

Reviewer 2: ```suggestion
Docker file is available [here](https://github.com/percona/pg_tde/blob/main/docker/Dockerfile)
```

Author: Fixed

_See [Make Builds for Developers](https://github.com/percona/pg_tde/wiki/Make-builds-for-developers) for more info on the build infrastructure._

## Setting up

>   1. Add extension to `shared_preload_libraries`:

Reviewer 2: ```suggestion
  1. Add extension to the `shared_preload_libraries`:
```

Author: Fixed
